Monika After Story/game/zz_submods.rpy

```diff
@@ -0,0 +1,307 @@
+        submod
+        for submod in store.mas_submod_utils.submod_map.values()
+        if submod.settings_pane is not None
+    ]) > 0
+
+init -991 python in mas_submod_utils:
+    import store
+    import sys
+    import traceback
+
+    persistent = store.persistent
+
+    submod_map = dict()
+
+    class SubmodError(Exception):
+        def __init__(self, _msg):
+            self.msg = _msg
+        def __str__(self):
+            return self.msg
+
+    class Submod:
+        """
+        Submod class
+
+        PROPERTIES:
+            author - submod author
+            name - submod name
+            description - submod description
+            version - version of the submod installed
+            dependencies - dependencies required for the submod
+            settings_pane - string referring to the screen used for the submod's settings
+            version_updates - update labels
+        """
+        def __init__(
+            self,
+            author,
+            name,
+            description,
+            version,
+            dependencies={},
+            settings_pane=None,
+            version_updates={}
+        ):
+            """
+            Submod object constructor
+
+            IN:
+                author - string, author name.
+
+                name - submod name
+
+                description - a short description for the submod
+
+                version - version number in format SPECIFICALLY like so: `1.2.3`
+                    (You can add more or less as need be, but splits MUST be made using periods)
+
+                dependencies - dictionary in the following structure: {"name": ("minimum_version", "maximum_version")}
+                corresponding to the needed submod name and version required
+                NOTE: versions must be passed in the same way as the version property is done
+                    (Default: empty dict)
+
+                settings_pane - a string representing the screen for this submod's settings
+
+                version_updates - dict of the format {"old_version_update_label_name": "new_version_update_label_name"}
+                    NOTE: submods MUST use the format <author>_<name>_v<version> for update labels relating to their submods
+                    NOTE: capital letters will be forced to lower and spaces will be replaced with underscores
+                    NOTE: Update labels MUST accept a version parameter, defaulted to the version of the label
+                    For example:
+                        author name: MonikaAfterStory,
+                        submod name: Example Submod
+                        submod vers: 1.2.3
+
+                    becomes:
+                        label monikaafterstory_example_submod_v1_2_3(version="v1_2_3")
+            """
+            #First make sure this name us unique
+            if name in submod_map:
+                raise SubmodError("A submod with name '{0}' already exists.\n".format(name))
```

Comment:
I think if this type of error is reached, we should just log the error and quietly ignore the submod. Especially since the likely user to encounter this would be a player installing a submod rather than the submod creator.